### PR TITLE
tsp, convenience api skip param of constant value

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**

### CODE REVIEW

File: typespec-tests/tsp/flatten.tsp [hunk 1]
```diff
@@ -101,7 +101,12 @@ model TodoItemPatch {
 interface FlattenOp {
   @route("send")
   @post
-  send(@query id: string, ...Request, ...ApiVersionParameter): OkResponse;
+  send(
+    @query id: string,
+    @query constantQueryParam: "constant",
```
Comment (weidongxu-microsoft) [inline typespec-tests/tsp/flatten.tsp:106]: Add a case of constant param. Verify it is set in Impl.

File: typespec-extension/src/code-model-builder.ts [hunk 2]
```diff
@@ -1123,7 +1123,7 @@ export class CodeModelBuilder {
         },
         language: {
           default: {
-            serializedName: param.name,
+            serializedName: this.getSerializedName(param.param),
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/code-model-builder.ts:1126]: After TCGC, we should not use this.

File: typespec-tests/src/main/java/com/cadl/flatten/implementation/FlattenClientImpl.java [hunk 3]
```diff
@@ -302,8 +304,9 @@ Response<Void> uploadTodoSync(@HostParam("endpoint") String endpoint,
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<Void>> sendWithResponseAsync(String id, BinaryData sendRequest,
         RequestOptions requestOptions) {
+        final String constantQueryParam = "constant";
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/main/java/com/cadl/flatten/implementation/FlattenClientImpl.java:307]: constantQueryParam = constant

File: typespec-extension/src/code-model-builder.ts [hunk 4]
```diff
@@ -1409,7 +1409,8 @@ export class CodeModelBuilder {
             // parameter
             if (
               existParameter.implementation === ImplementationLocation.Method &&
-              (existParameter.origin?.startsWith("modelerfour:synthesized/") ?? true)
+              (existParameter.origin?.startsWith("modelerfour:synthesized/") ?? true) &&
+              !(existParameter.schema instanceof ConstantSchema)
             ) {
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/code-model-builder.ts:1414]: Bug was here.

Now we just don't add the param to convenience API if it is constant.

**Review by XiaofeiCao — APPROVED** [review]

File: typespec-extension/src/code-model-builder.ts (hunk 2 above)
Comment (haolingdong-msft) [reply typespec-extension/src/code-model-builder.ts:1126]: I will try to rewrite this part to use TCGC's `(SdkQueryParameter | SdkPathParameter | SdkHeaderParameter)` types and use the `serializedName` on the type.

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**